v2/client/client_test.go

```diff
@@ -401,6 +403,71 @@ func TestClientContext(t *testing.T) {
 	wg.Wait()
 }
 
+func TestClientStartReceiverWithAckMalformedEvent(t *testing.T) {
+	testCases := []struct {
+		name        string
+		opts        []client.Option
+		expectedAck bool
+	}{
+		{
+			name: "without ack",
+		},
+		{
+			name:        "with ack",
+			opts:        []client.Option{client.WithAckMalformedEvent()},
+			expectedAck: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// make sure the receiver goroutine is closed
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			receiver := &mockReceiver{
+				finished: make(chan struct{}),
+			}
+
+			// only need 1 goroutine to exercise this
+			tc.opts = append(tc.opts, client.WithPollGoroutines(1))
+
+			c, err := client.New(receiver, tc.opts...)
+			if err != nil {
+				t.Errorf("failed to construct client: %v", err)
+			}
+
+			go c.StartReceiver(ctx, func(ctx context.Context, e event.Event) protocol.Result {
+				t.Error("receiver callback called unexpectedly")
+				return nil
+			})
+
+			// wait for receive to occur
+			time.Sleep(time.Millisecond)
```

Comment:
Removed.

I think I added it in a previous iteration that lacked the `select` below. And then `go test -race` run by CI failed the test run because it was indeed a bit flaky, and I forgot to remove it.